docs/rfcs/rfc-007-subscriptions.md

```diff
@@ -434,6 +960,122 @@ this0 { .name, actedIn: [ (this0)-[:ACTED_IN]->(this0_actedIn:Movie)   | this0_a
+            <<<
+            DELETE this_disconnect_actors0_rel
+            >>>
+            RETURN collect(meta) as disconnect_meta
+            <<<
+    }
+    >>>
+    WITH disconnect_meta + meta AS meta
+    WITH collect(meta) AS disconnect_meta
+    RETURN REDUCE(m=[],m1 IN disconnect_meta | m+m1 ) as disconnect_meta
+    <<<
+}
+WITH *
+WITH *, meta + disconnect_meta AS meta
+UNWIND meta AS m
+RETURN collect(DISTINCT this { .title }) AS data, collect(DISTINCT m) as meta
+```
+
+With a delete:
+
+-   Delete
+
+```cypher
+WITH [] AS meta
+MATCH (this:`Movie`)
+WHERE this.title = $param0
+WITH this, meta + { event: "delete", id: id(this), properties: { old: this { .* }, new: null }, timestamp: timestamp(), typename: "Movie" } AS meta
+>>>>>>>>>>>>>>>>>>>>>
+OPTIONAL MATCH (this)<-[this_relationship:*]-(this_destination0)
+WITH this, meta, collect(DISTINCT ???? this_destination0) as this_destination0_to_delete
+WITH this, this_destination0_to_delete, REDUCE(m=meta, n IN this_destination0_to_delete | m + { event: "delete", id: id(n), properties: { old: n { .* }, new: null }, timestamp: timestamp(), typename: ???? }) AS meta
+WITH this, this_destination0_to_delete, REDUCE(m=meta, n IN this_destination0_to_delete | m + { event: "disconnect", id_from: id(n), id_to: id(this), id: id(this_relationship), properties: {from: n {.*}, to: this {.*}, relationship: this_relationship {.*}} ,timestamp: timestamp(), relationshipName: ???? }) AS meta
+
+ RETURN count(*) AS _ // fix cardinality
+
+<<<<<<<<<<<<<<<<<<<<<
+DETACH DELETE this
+WITH meta
+UNWIND meta AS m
+RETURN collect(DISTINCT m) AS meta
+```
+
+-   Delete - delete
+
+1. Legacy foreach impl version
+
+```cypher
+WITH [] AS meta
+MATCH (this:`Movie`)
+WHERE this.title = $param0
+WITH this, meta + { event: "delete", id: id(this), properties: { old: this { .* }, new: null }, timestamp: timestamp(), typename: "Movie" } AS meta
+WITH this, meta
+OPTIONAL MATCH (this)<-[this_actors0_relationship:ACTED_IN]-(this_actors0:Actor)
+WHERE this_actors0.name = $this_deleteMovies_args_delete_actors0_where_Actorparam0
+WITH this, meta, collect(DISTINCT this_actors0) as this_actors0_to_delete
+WITH this, this_actors0_to_delete, REDUCE(m=meta, n IN this_actors0_to_delete | m + { event: "delete", id: id(n), properties: { old: n { .* }, new: null }, timestamp: timestamp(), typename: "Actor" }) AS meta
+>>>
+WITH this, this_actors0_to_delete, REDUCE(m=meta, n IN this_actors0_to_delete | m + { event: "disconnect", id_from: id(n), id_to: id(this), id: id(this_actors0_relationship), properties: {from: n {.*}, to: this {.*}, relationship: this_actors0_relationship {.*}} ,timestamp: timestamp(), relationshipName: "ACTED_IN" }) AS meta
+<<<
+CALL {
+        WITH this_actors0_to_delete
+        UNWIND this_actors0_to_delete AS x
+        DETACH DELETE x
+        RETURN count(*) AS _
+}
+DETACH DELETE this
+WITH meta
+UNWIND meta AS m
+RETURN collect(DISTINCT m) AS meta
+```
+
+2. Uniform with disconnect impl version
```

Comment:
@angrykoala would like your input on this: any preferences for either version 1 or 2?